src/tactic/ui/container/tab_wdg.py

```diff
@@ -844,27 +845,52 @@ def get_onload_js(my):
         spt.error('Tab close cannot find the header or content. Abort');
         return;
     }
-
     
-    var opener = header.getAttribute("spt_tab_opener");
-    var element_name = header.getAttribute("spt_element_name");
-    header.destroy();
+    var changed_element = false;
+    var changed_type = false;
+    function ok() {
+        //Remove unsaved changes flags
+        if (changed_element) {
+             changed_element.removeClass(changed_type);
+        }
 
-    content.destroy();
+        var opener = header.getAttribute("spt_tab_opener");
+        var element_name = header.getAttribute("spt_element_name");
+        header.destroy();
+
+        content.destroy();
 
-    var last_element_name = spt.tab.get_last_selected_element_name();
-    last_element_name = null;
+        var last_element_name = spt.tab.get_last_selected_element_name();
+        last_element_name = null;
 
-    // make the opener active
-    if (opener) {
-        spt.tab.select(opener);
+        // make the opener active
+        if (opener) {
+             spt.tab.select(opener);
+        }
+        else if (last_element_name) {
+            spt.tab.select(last_element_name);
+        }
+        else {
+            var last = headers[headers.length - 1].getAttribute("spt_element_name");
+            spt.tab.select(last);
+        }
+    }
+   
+    var changed_el = content.getElement(".spt_has_changes");
+    var changed_row = content.getElement(".spt_row_changed");
+    
+    if (changed_el) {
+        changed_element = changed_el;
+        changed_type = "spt_has_changed";
+        spt.confirm("There are unsaved changes in the current tab. Continue without saving?", ok, null);
     }
-    else if (last_element_name) {
-        spt.tab.select(last_element_name);
+    else if (changed_row) {
+        changed_element = changed_row;
+        changed_type = "stp_row_changed";
+        spt.confirm("There are unsaved changes in the current tab. Continue without saving?", ok, null);
     }
     else {
-        var last = headers[headers.length - 1].getAttribute("spt_element_name");
-        spt.tab.select(last);
+       ok();
     }
 }
```

Comment:
I would remove changed_element and changed_type variables if they are not being used.. They are declared near the top as well. 
Can the message tell the user what kind of unsaved changes he has? like in tables or in UI